cogs/source/source.py

```diff
@@ -23,37 +25,46 @@ def messageBuilder(titleEnglish: str, anilistID: str, episode: str, similarity:
     return message
 
 
-async def postSourceFunction(ctx, imageURL):
-    """helper method"""
-    try:
-        # use the API to get results
+class Source(commands.Cog):
 
-        tracemoe = tracemoepy.AsyncTrace()
-        result = await tracemoe.search(imageURL.strip("<>"), is_url=True)
-        titleEnglish = result.result[0].anilist.title.english or "No Title Found"
-        anilistID = f"{result.result[0].anilist.id}" or "No anilistID Found"
-        episode = f"{result.result[0].episode}" or "No Episode Found"
-        similarity = float(result.result[0].similarity) or 0
-        await tracemoe.aio_session.close()
+    tracemoe = None
 
-        # send the message using messageBGuilder to build the message
-        await ctx.send(messageBuilder(titleEnglish, anilistID, episode, similarity))
+    def __init__(self, bot: Red):
+        self.tracemoe = tracemoepy.AsyncTrace()
 
-    except TooManyRequests:
-        await ctx.send("Please try again later")
-    except EntityTooLarge:
-        await ctx.send("Too big of file image")
-    except ServerError:
-        await ctx.send(
-            "Server error. Ensure image is provided as URL and points directly to png or jpg image"
-        )
-    except EmptyImage:
-        await ctx.send(
-            "Empty image provided. Ensure image is provided as URL and points directly to png or jpg image"
-        )
+    async def __unload(self):  # pylint: disable=invalid-name
+        await self.tracemoe.aio_session.close()
```

Comment:
Let's not copy the function name from the other file; can we use a better name? Also add logging to help us double check that we end up here.
```suggestion
    async def closeSession(self):
        self.logger.debug("Closing session")
        await self.tracemoe.aio_session.close()
```